src/headless/plugins/vcard/utils.js

```diff
@@ -40,12 +40,13 @@ export function createStanza(type, jid, vcard_el) {
  */
 export function onOccupantAvatarChanged(occupant) {
     const hash = occupant.get('image_hash');
+    const url = occupant.get('image_url');
     const vcards = [];
     if (occupant.get('jid')) {
         vcards.push(_converse.state.vcards.get(occupant.get('jid')));
     }
     vcards.push(_converse.state.vcards.get(occupant.get('from')));
-    vcards.forEach((v) => hash && v && v?.get('image_hash') !== hash && api.vcard.update(v, true));
+    vcards.forEach((v) => (hash || url) && v && (v?.get('image_hash') !== hash || v?.get('image_url') !== url) && api.vcard.update(v, true));
```

Comment:
This still doesn't look right, because of the `(hash || url)`  the `hash` value could be undefined, but then the `v?.get('image_hash') !== hash` check could still execute (because `url` is defined), potentially resulting in an unnecessary call to `api.vcard.update`.

I think it's better to write this logic out.


```suggestion
    vcards.filter((v) => v).forEach((v) => {      
        if (hash && v.get('image_hash') !== hash || url && v.get('image_url') !== url) {
            api.vcard.update(v, true);
        }
    });
```